test/org/sagebionetworks/bridge/models/schedules/TaskSchedulerTest.java

```diff
@@ -0,0 +1,168 @@
+package org.sagebionetworks.bridge.models.schedules;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.sagebionetworks.bridge.models.schedules.ScheduleTestUtils.dt;
+import static org.sagebionetworks.bridge.models.schedules.ScheduleTestUtils.assertDates;
+import static org.sagebionetworks.bridge.models.schedules.ScheduleType.ONCE;
+
+import java.util.List;
+import java.util.Map;
+
+import org.joda.time.DateTime;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.newrelic.agent.deps.com.google.common.collect.Maps;
+
+/**
+ * These tests cover other aspects of the scheduler besides the accuracy of its scheduling.
+ *
+ */
+public class TaskSchedulerTest {
```

Comment:
Other unit tests I feel we should add:

* No event found - Events is an empty map. No enrollment or anything. From reading the code, it looks like the scheduler would return an empty list, but I didn't see a unit test for this.

* A test encapsulating something like "1 hour after breakfast".